portlet-parent/wicketstuff-portlet/src/main/java/org/apache/wicket/portlet/request/mapper/MapperDelegate.java

```diff
@@ -0,0 +1,105 @@
+package org.apache.wicket.portlet.request.mapper;
+
+import org.apache.wicket.Application;
+import org.apache.wicket.core.request.handler.PageProvider;
+import org.apache.wicket.core.request.handler.RenderPageRequestHandler;
+import org.apache.wicket.core.request.mapper.IPageSource;
+import org.apache.wicket.protocol.http.PageExpiredException;
+import org.apache.wicket.protocol.http.WebApplication;
+import org.apache.wicket.request.IRequestHandler;
+import org.apache.wicket.request.Url;
+import org.apache.wicket.request.Url.QueryParameter;
+import org.apache.wicket.request.component.IRequestablePage;
+import org.apache.wicket.request.mapper.info.PageInfo;
+import org.apache.wicket.request.mapper.parameter.IPageParametersEncoder;
+import org.apache.wicket.request.mapper.parameter.PageParameters;
+import org.apache.wicket.util.lang.Args;
+
+/**
+ * Handles the processHybrid and encodePageParameters requests of {@link BookmarkableMapper} and {@link HomePageMapper} 
+ */
+public class MapperDelegate
+{
+
+	/**
+	 * Creates a {@code IRequestHandler} that processes a hybrid request. When the page identified
+	 * by {@code pageInfo} was not available, the request should be treated as a bookmarkable
+	 * request.
+	 * 
+	 * @param pageInfo
+	 * @param pageClass
+	 * @param pageParameters
+	 * @param renderCount
+	 * @return a {@code IRequestHandler} capable of processing the hybrid request.
+	 */
+	protected IRequestHandler processHybrid(PageInfo pageInfo,
+		Class<? extends IRequestablePage> pageClass, PageParameters pageParameters,
+		Integer renderCount)
+	{
+		PageProvider provider = new PageProvider(pageInfo.getPageId(), pageClass, pageParameters,
+			renderCount);
+		provider.setPageSource((IPageSource)Application.get().getMapperContext());
+		if (provider.isNewPageInstance() &&
+			!WebApplication.get().getPageSettings().getRecreateBookmarkablePagesAfterExpiry())

```

Comment:
You can cache the value of `Application.get()` into a local variable to avoid making two ThreadLocal lookups.